packages/near-membrane-base/src/membrane.ts

```diff
@@ -479,6 +481,14 @@ export function createMembraneMarshall() {
                 targetTraits |= TargetTraits.IsArray;
             } else if (!targetIsFunction) {
                 targetTraits |= TargetTraits.IsObject;
+                const targetProto = ReflectGetPrototypeOf(target);
```

Comment:
Can we link to where we base our implementation on to show it's a explicit design choice.

Reply:
Yes